src/LUtility/Types/Quaternion.h

```diff
@@ -0,0 +1,105 @@
+#ifndef LEPUS_UTILITY_TYPES_QUATERNION
+#define LEPUS_UTILITY_TYPES_QUATERNION
+
+#include "Vector.h"
+#include "../../LEngine/ConsoleLogger.h"
+#include <cmath>
+#include "../Math.h"
+
+namespace lepus
+{
+    namespace types
+    {
+        class Quaternion : public Vector4
+        {
+            public:
+            Quaternion()
+            {
+                const float identity[] = { 0.f, 0.f, 0.f, 1.f };
+                init((float*)identity);
+            }
+
+            Quaternion(const Quaternion& quat)
+            {
+                init(quat);
+            }
+
+            Quaternion(float* quatData)
+            {
+                init(quatData);
+            }
+
+            Quaternion(float axisX, float axisY, float axisZ, float angle)
```

Comment:
add summary clarifying that this converts parameters to their quaternion part representations - users shouldn't assume this is just for initialising xyzw